demos/paywall_bot/demo.py

```diff
@@ -0,0 +1,141 @@
+    "<b>Main Menu</b> screen even after typing the /start command. "
+    "\n"
+    "<i>If you'd like to try the demo again, just click the button below "
+    "to make a fake refund 👇</i>"
+)
+
+FAKE_PAYMENT_SCREEN_DESCRIPTION = (
+    'To continue using the bot, you need to make a <i>fake</i> payment.\n'
+    '\n'
+    'Do you want to continue?'
+)
+
+PAYMENT_SCREEN_DESCRIPTION = (
+    "Welcome to HammettPaywallBot!\n"
+    "\n"
+    "Now you see the <b>Payment</b> screen, and you <i>won't see any other screens</i> until "
+    "you make a <i>fake</i> payment. So, if you type the /start command, "
+    "you'll just get this screen again."
+)
+
+
+class FakePaymentScreen(Screen):
+    """The class implements FakePaymentScreen."""
+
+    description = FAKE_PAYMENT_SCREEN_DESCRIPTION
+
+    async def add_default_keyboard(self, _update, _context):
+        """Set up the keyboard for the screen."""
+        return [[
+            Button(
+                '❌ No',
+                PaymentScreen,
+                source_type=SourceTypes.MOVE_SOURCE_TYPE),
+            Button(
+                '✅ Yes',
+                self.handle_fake_payment,
+                source_type=SourceTypes.HANDLER_SOURCE_TYPE),
+        ]]
+
+    @ignore_permissions([PaywallPermission])
+    @register_button_handler
+    async def handle_fake_payment(self, update, context):
+        """Handle a button click and process a fake payment.
+
+        It is better to decorate all the handlers which are used
+        in a permission denied state.
+        """
+        user = update.effective_user
+        settings.PAID_USERS.append(user.id)
+
+        return await MainMenuScreen().move(update, context)
+
+    @ignore_permissions([PaywallPermission])
+    async def move(self, update, context, **kwargs):
+        """Switch to the screen re-rendering the previous message.
+
+        It is better to decorate all the handlers which are used
+        in a permission denied state.
+        """
+        return await super().move(update, context, **kwargs)
+
+
+class MainMenuScreen(StartMixin):
+    """The class implements MainMenuScreen that is only available after the payment."""
+
+    description = MAIN_MENU_SCREEN_DESCRIPTION
+
+    async def add_default_keyboard(self, _update, _context):
+        """Set up the keyboard for the screen."""
+        return [[
+            Button(
+                '💸 Fake Refund',
+                self.handle_fake_refund,
+                source_type=SourceTypes.HANDLER_SOURCE_TYPE),
+        ]]
+
+    @register_button_handler
+    async def handle_fake_refund(self, update, context):
+        """Handle a button click and do fake refund."""
+        settings.PAID_USERS = []
+
+        return await PaymentScreen().move(update, context)
+
+
+class PaymentScreen(Screen):
+    """The class implements PaymentScreen, which acts as a response
+    if the user has not paid.
+    """
```

Comment:
`The class implements PaymentScreen, which appears when the user hasn't made a payment.`